streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -509,11 +509,11 @@ void to(final TopicNameExtractor<K, V> topicExtractor,
      *
      * KStream outputStream = inputStream.transform(new TransformerSupplier() { ... }, "myTransformState");
      * }</pre>
-     * Within the {@link Transformer}, the state is obtained via the {@link  ProcessorContext}.
+     * Within the {@link Transformer}, the state is obtained via the {@link ProcessorContext}.
      * To trigger periodic actions via {@link org.apache.kafka.streams.processor.Punctuator#punctuate(long) punctuate()},
      * a schedule must be registered.
      * The {@link Transformer} must return a {@link KeyValue} type in {@link Transformer#transform(Object, Object)
-     * transform()} and {@link org.apache.kafka.streams.processor.Punctuator#punctuate(long) punctuate()}.
+     * transform(K, V)} and {@link org.apache.kafka.streams.processor.Punctuator#punctuate(long) punctuate(long)}.
```

Comment:
We don't add parameters in any JavaDocs. We use `()` in purpose to "shorten" long parameter lists (if usr follows the link, she sees the full parameter list). Note that
```
{@link org.apache.kafka.streams.processor.Punctuator#punctuate(long)}
```
and
```
{@link org.apache.kafka.streams.processor.Punctuator#punctuate(long) punctuate(long)}
```
would be the same (both render as `punctuate(long)`), but we want to get simplified `punctuate()`.

Similar below.

Reply:
I removed the parameters from the links in the parts that I modified. However, in the other parts there are also links with parameters, e.g., line 544. Should I also change those in this PR?

Reply:
I was just looking across the JavaDocs. Seems we are not very consistent in what we do...  L544 is not the only example. I leave it up to you if you want to clean it up, or not. We also just merge this PR as-is. Might not be worth the time?

Reply:
I removed the parameters from the links to methods in the javadocs for `transform` and `flatTransform`.